compiler/rustc_hir_analysis/src/check/mod.rs

```diff
@@ -546,3 +554,80 @@ fn bad_non_zero_sized_fields<'tcx>(
 pub fn potentially_plural_count(count: usize, word: &str) -> String {
     format!("{} {}{}", count, word, pluralize!(count))
 }
+
+pub fn check_function_signature<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    cause_code: ObligationCauseCode<'tcx>,
```

Comment:
Can you make the caller pass in a full `ObligationCause`? That's usually more conventional than just passing the code.